ui/src/main/js/field/input/TableSelectInput.js

```diff
@@ -72,51 +75,30 @@ class TableSelectInput extends Component {
         this.state = {
             progress: false,
             showTable: false,
+            currentPage: 1,
+            currentPageSize: 10,
+            totalPageCount: 0,
+            currentSearchTerm: '',
             data: [],
             selectedData: [],
             displayedData: [],
             showClearConfirm: false
         };
     }
 
-    // FIXME componentWillMount is deprecated, use the constructor or componentDidMount to initialize this
-    componentWillMount() {
-        const { value } = this.props;
-        if (value && value.length > 0) {
-            this.retrieveTableData().then(() => {
-                this.updateSelectedValues();
-            });
-        }
-    }
-
-    componentDidUpdate(prevProps) {
-        const { value } = this.props;
-        const prevSize = prevProps.value && prevProps.value.length === 0;
-        const currentSize = value && value.length > 0;
-        const emptySelected = this.state.selectedData.length === 0;
-        if (prevSize && currentSize && emptySelected) {
-            if (this.state.data.length == 0) {
-                this.retrieveTableData().then(() => {
-                    this.updateSelectedValues();
-                });
-            } else {
-                this.updateSelectedValues();
-            }
-        }
+    componentDidMount() {
+        this.updateSelectedValues();
     }
 
     updateSelectedValues() {
         const { value } = this.props;
-        const { selectedData, data } = this.state;
+        const { selectedData } = this.state;
         selectedData.push(...value);
-        const keyColumnHeader = this.props.columns.find((column) => column.isKey).header;
         const convertedValues = selectedData.map((selected) => {
-            const columnContainsValue = data.map((dataValue) => dataValue[keyColumnHeader])
-                .includes(selected);
             return {
                 label: selected,
                 value: selected,
-                missing: !columnContainsValue
+                missing: false
```

Comment:
We cannot support "missing" data with paging because the data could be in a different page that has yet to be retrieved.